docs/connectors/sources/kafka.md

```diff
@@ -52,9 +51,11 @@ CREATE TABLE INPUT (
       "transport": {
           "name": "kafka_input",
           "config": {
-              "bootstrap.servers": "example.com:9092",
-              "auto.offset.reset": "earliest",
-              "topics": ["book-fair-sales"]
+              "topic": "book-fair-sales",
+              "start_from": "earliest",
+              "kafka_options": {
+                  "bootstrap.servers": "example.com:9092"
```

Comment:
I'm having second thoughts about nesting rdkafka options. First input and output Kafka configs are no longer symmetric. Second, `bootstrap.servers` is really a mandatory parameter and the most important field in the Kafka connector config, so nesting it inside another struct seems counterintuitive. Third, why should the user care that some of these parameters are interpreted by Feldera, while other are  passed to some library we use? From their perspective they are all just part of the Kafka transport config. Finally nested JSONs are a common source of user errors, so adding more nesting will likely make it worse.

Reply:
The big reason why I want to separate them is to enable validating the configuration before we pass it to librdkafka. A misspelled configuration meant for us will result in a weird error from librdkafka.

However, after thinking about it for a bit, I think that we can distinguish the two kinds of configuration options: the ones for librdkafka almost all contain `.` (and we can list the exceptions), and the ones for us never do. I can use that.

> Third, why should the user care that some of these parameters are interpreted by Feldera, while other are passed to some library we use? 

They have to know that anyway because we're not going to document the librdkafka options ourselves, there are too many of them.

Reply:
Sounds good, thanks!

I actually think the right move would be to one day document the relevant features rather than refer the user to rdkafka docs. I once went through all of them. Not all rdkafka features make sense for us (e.g., there are things like callbacks, consumer-group-related options are invalid for us, etc.).  You don't need to do this now of course, just saying that the ideal UX wouldn't expose the existence of rdkafka to most users.

Reply:
> the ideal UX wouldn't expose the existence of rdkafka to most users.

I agree